Assets/MixedRealityToolkit/Services/BaseCoreSystem.cs

```diff
@@ -0,0 +1,30 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.MixedReality.Toolkit.Core.Definitions;
+using Microsoft.MixedReality.Toolkit.Core.Interfaces;
+
+namespace Microsoft.MixedReality.Toolkit.Core.Services
+{ 
+    public abstract class BaseCoreSystem : BaseEventSystem
+    {
+        public BaseCoreSystem(
+            IMixedRealityServiceRegistrar registrar,
+            BaseMixedRealityProfile profile = null) : base()
+        {
+            Registrar = registrar;
+            ConfigurationProfile = profile;
```

Comment:
I think we need to support this being null. I think that in many instances there aren't null checks using this Registrar.

Reply:
good call